### PR TITLE
client/web: when tracing a query, add a link to the trace

### PR DESCRIPTION
**Pull request by rmmh**
This only works for streaming queries. The trace URL is passed along in
the streaming query's progress event because the DOM EventSource API
doesn't support reading response headers for the X-Trace option. As a
benefit, being part of the normal data pipeline results in cleaner code.

The new "View Trace" link is just to the right of the query progress summary:
image

### CODE REVIEW

**Review by keegancsmith — COMMENTED** [review]
Nice stuff! The backend code LGTM. I'll leave to @limitedmage to approve. Some general comments though that will likely require an update to the PR:

- some sites always have tracing enabled / sample. We don't want to always show the trace link in that case.
- why copy to paste board? The user should be able to just click on the link?

Maybe a fix for the first point is to only send down the trace link if the user is an admin. Although I suppose the idea is for users to share the link with admins if the opt-in?

**Comment by rmmh** [comment]
Maybe it should only show the tracing link if `&trace=1` is in the URL? That might be a better compromise.

It doesn't copy to the clipboard-- that code just shows an icon of a clipboard with the trace symbol, as you can see in the screenshot.

**Comment by rmmh** [comment]
OK, I made it so it only displays this link when `&trace=1` is in the URL. This ensures that we only show the link when a user is expecting it by opting in.

File: client/web/src/search/results/streaming/progress/StreamingProgressCount.test.tsx [hunk 1]
```diff
@@ -75,4 +75,25 @@ describe('StreamingProgressCount', () => {
...
+        const progress: Progress = {
+            durationMs: 0,
+            matchCount: 0,
+            skipped: [],
+            trace: 'https://sourcegraph.test:3443/-/debug/jaeger/trace/abcdefg',
+        }
+
+        expect(mount(<StreamingProgressCount state="loading" progress={progress} showTrace={true} />)).toMatchSnapshot()
+    })
+    it('should not render a trace link when not opted into with &trace=1', () => {
```
Comment (limitedmage) [inline client/web/src/search/results/streaming/progress/StreamingProgressCount.test.tsx:89]:
```suggestion
    })

    it('should not render a trace link when not opted into with &trace=1', () => {
```

**Review by limitedmage — APPROVED** [review]
Frontend code looks great!

**Comment by limitedmage** [comment]
One more suggestion: would be nice to add a storybook test for when the icon is shown.

**Pull request merged**